cmd/fetch.go

```diff
@@ -35,3 +38,85 @@ func init() {
 	_ = fetchCmd.Flags().MarkHidden("redact-diags")
 	rootCmd.AddCommand(fetchCmd)
 }
+
+// filterConfigProviders gets a list of "providerAlias:resource1,resource2" items and updates the given config, removing non-matching providers
+// valid usages:
+// "aws" or "aws:*" (all resources)
+// "aws:ec2.instances,s3.buckets" (only ec2.instances and s3.buckets)
+func filterConfigProviders(list []string) func(*config.Config) error {
```

Comment:
I am still not a fan of this implementation, as it feels harder, also its limited as it will be harder to upgrade it for example I can't do aws:ec2.* for example.

I think the following should be done -> parse fetch arg into a structure i.e RequestedProviders.
-> pass this to the fetch command. 
-> in fetch would loop over requested providers if its empty would loop over all providers in config. 
 ---> if requested providers is not empty, for each provider then get the provider from config, if we also requested resources swap the resources in the fetch request.